src/main/java/io/github/moulberry/notenoughupdates/NEUOverlay.java

```diff
@@ -1907,27 +1854,18 @@ private ItemStack getWardrobeSlot(int armourSlot) {
 		} else return null;
 	}
 
-	public ItemStack slot1 = null;
-	public ItemStack slot2 = null;
-	public ItemStack slot3 = null;
-	public ItemStack slot4 = null;
-	public ItemStack petSlot = null;
-
-	public static boolean isRenderingArmorHud() {
-		return renderingArmorHud;
-	}
-
-	public static boolean isRenderingPetHud() {
-		return renderingPetHud;
-	}
-
 	/**
 	 * Renders the search bar, quick commands, item selection (right), item info (left) and armor hud gui elements.
 	 */
 	public void render(boolean hoverInv) {
 		if (disabled) {
 			return;
 		}
+		if (Keyboard.isRepeatEvent()) {
+			keyboardInput(false);
+			Utils.sleepOnCurrentThread(15);
```

Comment:
Why are we sleeping on the rendering thread. Even for just 15ms that seems like a bad idea

Reply:
Without that the characters will be removed/added really really fast and I didn't notice any issues with it

Reply:
I feel like you are slowing down the fps so we dont type too fast? At least i think thats whtat you are doing

Reply:
Wait why are we processing userinput in the render event in the first place??